statefulset/scale.go

```diff
@@ -0,0 +1,92 @@
+package statefulset
+
+import (
+	"fmt"
+	"strconv"
+)
+
+// ReplicasAnnotation stores the initial number of replicas before scaling down the StatefulSet.
+const ReplicasAnnotation = "sts-resize.appuio.ch/replicas"
+
+// ScalupAnnotation marks a replica as in the process of scaling back up and prevents the controller from scaling it down.
+const ScalupAnnotation = "sts-resize.appuio.ch/scalup"
+
+// ScaleDown scales down the replica to 0, saves the original state and returns true if it ran successfully before
+func (s *Info) ScaleDown() bool {
+	if s.scaledDown() || s.isScalingUp() {
+		return true
+	}
+	s.saveOriginalReplicas()
+	r := int32(0)
+	s.sts.Spec.Replicas = &r
+	return false
+}
+
+// ScaleUp scales back up to the original scale.
+// Returns true if it ran successfully before
+func (s Info) ScaleUp() (bool, error) {
+	scale, err := s.getOriginalReplicas()
+	if err != nil {
```

Comment:
Unfortunate naming. This function doesn't _actually_ do any scaling, it just mutates the specs  accordingly.
What about `PrepareSpecForScaleUp`?